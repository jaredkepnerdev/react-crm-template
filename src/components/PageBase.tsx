import * as React from "react";
import Paper from "@material-ui/core/Paper";
import Divider from "@material-ui/core/Divider";
import { grey } from "@material-ui/core/colors";

const grey600 = grey["600"];


const styles={
  navigation: {
    fontSize: 15,
    fontWeight: 400, //TypographyStyle.fontWeightLight,
    color: grey600,
    paddingBottom: 15,
    display: "block",
  },
  title: {
    fontSize: 24,
    fontWeight: 500, // TypographyStyle.fontWeightLight,
    marginBottom: 20,
  },
  paper: {
    padding: 10,
  },
  main:{
    paddingTop: 80,
    paddingLeft: 30,
    paddingRight: 30
  },
  clear: {
    clear: "both" as TODO
  }
}

interface PageBaseProps  {
  title: string,
  navigation: string,
  children: React.ReactChild
};



const PageBase: React.FC<PageBaseProps> = 
({ title, navigation, children }) => {
  return (
    <div style={styles.main}>
      <span style={styles.navigation}>{navigation}</span>

      <Paper style={styles.paper}>
        <h3 style={styles.title}>{title}</h3>

        <Divider />
        {children}

        <div style={styles.clear} />
      </Paper>
    </div>
  );
};

export default PageBase;
