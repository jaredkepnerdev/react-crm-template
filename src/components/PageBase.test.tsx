import React from "react";
import { shallow } from "enzyme";
// import { expect } from "chai";
import PageBase from "./PageBase";
import {create, act} from 'react-test-renderer';

// describe("<PageBase />", () => {
//   it("should have a span with the navigation text", () => {
//     const wrapper = shallow(<PageBase navigation="navigation" />);
//     const actual = wrapper.find("span").text();
//     const expected = "navigation";

//     expect(expected).to.equal(actual);
//   });

//   it("should have a header with the title text", () => {
//     const wrapper = shallow(<PageBase title="title" />);
//     const actual = wrapper.find("h3").text();
//     const expected = "title";

//     expect(expected).to.equal(actual);
//   });
// });


test('renders learn react link', () => {
  let app;
  act(() => {
    app = create(
    <PageBase title="title" navigation="navigation">
      {"empty content"}
    </PageBase>
    );
  });

  expect(app.toJSON()).toMatchSnapshot();

});
