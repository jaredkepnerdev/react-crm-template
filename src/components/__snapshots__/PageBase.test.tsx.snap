// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  style={
    Object {
      "paddingLeft": 30,
      "paddingRight": 30,
      "paddingTop": 80,
    }
  }
>
  <span
    style={
      Object {
        "color": "#757575",
        "display": "block",
        "fontSize": 15,
        "fontWeight": 400,
        "paddingBottom": 15,
      }
    }
  >
    navigation
  </span>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <h3
      style={
        Object {
          "fontSize": 24,
          "fontWeight": 500,
          "marginBottom": 20,
        }
      }
    >
      title
    </h3>
    <hr
      className="MuiDivider-root"
    />
    empty content
    <div
      style={
        Object {
          "clear": "both",
        }
      }
    />
  </div>
</div>
`;
